### ISSUE TITLE
[UX] Update help text for URL aliases.

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#2061]
Current text: 
```
URL patterns automatically alias new content based on wildcards called "tokens". For example the URL "node/10" might be automatically aliased to "blog/my-first-post" using the pattern "blog/[node:title]".
```

Current proposed alternative (needs work):
```
Your site can have URLs generated automatically based on wildcards called <em>tokens</em> or <em>URL aliases</em> can be defined manually in the UI. For automatic generation of URL aliases, define <em>URL alias patterns</em> for each type of content below
```

Follow-up from https://github.com/backdrop/backdrop-issues/issues/1250#issuecomment-239713022

**Comment by klonos** [comment]
👍 

I think that an "either" is in order before "have URLs generated automatically". Also a comma after "tokens" (??).

How about this?:

`Backdrop can be configured to either automatically generate URL aliases for your content based on wildcards called <em>tokens</em>, or allow you to specify the URL aliases manually during content creation. To enable the automatic generation of URL aliases, define the <em>URL alias patterns</em> for each type of content below. Leaving the <em>URL alias pattern</em> empty, will allow manual creation of URL aliases for these content types instead.`

**Comment by klonos** [comment]
...a bit wordy, but to the point I believe.

**Comment by jenlampton** [comment]
I like that text for the top of the URL alias patterns page! 

It's a bit much for the content type creation form, but I think we can re-use some in places. I really liked the previous "for example" there because I often copy the token for [node:title] right from the desc instead of opening the token browser.

`Enable automatic generation of URL aliases by defining a pattern containing tokens. For example the URL "node/10" might be automatically aliased to "blog/my-first-post" using the pattern "blog/[node:title]".`

**Comment by klonos** [comment]
Too early in the morning and have to run, but initial reaction was this... my proposed text has three sentences:

1st explains the options
2nd explains what to do to enable automatic generation of URL aliases
3rd explains what to do for manual creation of URL aliases

...perhaps we could consolidate the text of explaining the options with each option and just have two bullets instead:

- To have URL aliases automatically generated ...
- To allow manual creation of URL aliases during edit...

Anyways, will get back to this at the end of the day.

👍 on using the example.

**Comment by quicksketch** [comment]
Thanks for splitting this minor issue out from https://github.com/backdrop/backdrop-issues/issues/1250#issuecomment-239713022 :smile:

**Comment by klonos** [comment]
OK, this is what I've come up with:

"To have URL aliases generated automatically for your content, you can specify patterns for each content type using wildcards called <em>tokens</em>. For example, the URL "node/10" might be automatically aliased to "blog/my-first-post" using the pattern "blog/[node:title]" (the token in this example is [node:title]).

Alternatively, leave the <em>URL alias pattern</em> empty in order to allow URL aliases for your content to be created manually during edit."

...**BUT**:

a) The text so far (and the example) is revolving around content types, but in that form we also allow patterns to be defined for taxonomy terms and user accounts. Should we accommodate for that?

b) If content authors do not have the "Create and edit URL aliases" permission, then the "...to allow URL aliases for your content to be created manually during edit." part of the help text does not apply. In fact, the site admin will leave patterns empty (believing that users will be allowed to manually add URL aliases according to help text), but because users won't have the right permission to define a URL alias, their content will end up with `/node/xyz` upon save. Should we have a "set proper permissions" link there (link text TBD) pointing to `/admin/config/people/permissions#edit-create-url-aliases` so that people do not hit this WTF?

c) In #940 we aim to change the default URL pattern for content types to `content/[node:title]` (we should - for parity with D7). If we implement that, then we solve point b above (sorta - there won't be any ugly `/node/xyz` paths at least). Additionally, we should adapt the help text to something like this:

"By default, all content created in your site automatically receives a URL alias in the form of `content/[node:title]` (you can change this pattern below). The `[node:title]` part of this URL alias pattern is called a *token* and it acts as a wildcard. When saving the content, the tokens are replaced with their actual value, which in this case is the content title. So, saving a page with title "About us", will result in the URL alias `content/about-us`.

All content types with an empty <em>URL alias pattern</em> below will receive the default `content/[node:title]` URL alias. To override this default URL alias for a specific content type, change its empty <em>URL alias pattern</em> with a custom one.

In order to allow URL aliases for your content to be created manually by content editors during edit, you will need to give the respective users the "[Create and edit URL aliases](url)" permission."

Thoughts?

**Comment by klonos** [comment]
BTW, I really like the approach that pathauto 8.x has taken with their new, redesigned UI (#2066).

In that, they have added this help text in `/admin/config/search/path`

"An alias defines a different name for an existing URL path - for example, the alias 'about' for the URL path 'node/1'. A URL path can have multiple aliases."

**Comment by jenlampton** [comment]
> In that, they have added this help text

Do you find that help text helpful?  I struggle with all the instances of URL path and alias, the words mix together and I don't really make sense of what the help text is trying to convey.

**Comment by jenlampton** [comment]
> The text so far (and the example) is revolving around content types

Yes, that is because we are adding this text to the content type form. :)

**Comment by Amulya-coder** [comment]
Hi, I would like to take up this issue, I am a contributor to OpenForce 2022. Thanks

**Comment by Amulya-coder** [comment]
Hi, I have created the PR for this issue: https://github.com/backdrop/backdrop/pull/4001
Kindly review my PR and if there are any required changes to be done, please let me know.
Thank you

**Comment by stpaultim** [comment]
@Amulya-coder - Thank you for this PR. As you can see, this issue is quite old and has been waiting for someone to take the step of creating an actual PR that others can respond to. Your PR has motivated me to weigh in on this issue for the first time and propose a slight modification to the earlier suggested text. 

I recommend that we keep this issue focused ONLY on this page: admin/config/urls/path/patterns (there was some suggestion that we might also be working on the content type creation page, I suggest we open a followup issue for that if necessary). 

My new proposal does the following:
1) Adds a clearer explanation of what URL aliases are up front. 
2) Keeps much of the text from @klonos 
3) Preserves the example that I understood @jenlampton likes. 

`URL aliases are human readable alternatives to the machine generated URLs that Backdrop creates by default. You can build these aliases using wildcards called tokens. For example, the URL "node/10" might be automatically aliased to "blog/my-first-post" using the pattern "blog/[node:title]" (the token in this example is [node:title]).`

@Amulya-coder - I think that your next steps are the following. 

1) Provide your own feedback as someone that is new to Backdrop, about whether or not this text changes helps you understand what this feature is about. Your input is very valuable.
2) Wait for 2-3 days and see what feedback we get and then update your PR based upon that feedback. Your current PR uses the initial proposed text by @jenlampton, but does not account for some of the later discussion (which is understandable, since it's very hard to follow)

Does anyone else have any feedback?

**Comment by stpaultim** [comment]
**ORIGINAL:**
URL patterns automatically alias new content based on wildcards called tokens. For example the URL node/10 might be automatically aliased to blog/my-first-post using the pattern blog/[node:title].

**JEN PROPOSED & CURRENT PR**
Your site can have URLs generated automatically based on wildcards called <em>tokens</em> or <em>URL aliases</em> can be defined manually in the UI. For automatic generation of URL aliases, define <em>URL alias patterns</em> for each type of content below

**GREG (2016/08/16):**
To have URL aliases generated automatically for your content, you can specify patterns for each content type using wildcards called tokens. For example, the URL "node/10" might be automatically aliased to "blog/my-first-post" using the pattern "blog/[node:title]" (the token in this example is [node:title]).

Alternatively, leave the URL alias pattern empty in order to allow URL aliases for your content to be created manually during edit.

**TIM (2022/03/18)**
URL aliases are human readable alternatives to the machine generated URLs that Backdrop creates by default. You can build these aliases using wildcards called tokens. For example, the URL "node/10" might be automatically aliased to "blog/my-first-post" using the pattern "blog/[node:title]" (the token in this example is [node:title]).
